pkg/trace/pb/hook_test.go

```diff
@@ -22,13 +22,28 @@ func TestMetaHook(t *testing.T) {
 		s, err := decodeBytes(b)
 
 		assert := assert.New(t)
+		assert.False(HasMetaHooks())
 		assert.Nil(err)
 		assert.Equal(map[string]string{"card.number": "4166 6766 6766 6746"}, s.Meta)
 	})
 
-	t.Run("on", func(t *testing.T) {
-		SetMetaHook(func(k, v string) string { return "test" })
-		defer SetMetaHook(nil)
+	t.Run("meta struct off", func(t *testing.T) {
+		b := newEmptyMessage()
+		b = msgp.AppendString(b, "meta_struct")
+		b = msgp.AppendMapHeader(b, 1)
+		b = msgp.AppendString(b, "appsec")
+		b = msgp.AppendBytes(b, msgp.AppendString(nil, "4166 6766 6766 6746"))
+		s, err := decodeBytes(b)
+
+		assert := assert.New(t)
+		assert.False(HasMetaHooks())
+		assert.Nil(err)
+		assert.Equal(map[string][]byte{"appsec": msgp.AppendString(nil, "4166 6766 6766 6746")}, s.MetaStruct)
+	})
+
+	t.Run("meta on", func(t *testing.T) {
```

Comment:
Same here. But ideally, you'd have 3 tests:
* meta hook only (`meta`)
* meta_struct hook only (`meta_struct`)
* both hooks (`both`)